test/core/client_config/resolvers/sockaddr_resolver_test.c

```diff
@@ -50,9 +67,19 @@ static void test_succeeds(grpc_resolver_factory *factory, const char *string) {
   args.uri = uri;
   resolver = grpc_resolver_factory_create_resolver(factory, &args);
   GPR_ASSERT(resolver != NULL);
+
+  on_resolution_arg on_res_arg;
+  memset(&on_res_arg, 0, sizeof(on_res_arg));
+  on_res_arg.expected_server_name = gpr_strdup(uri->path);
+  grpc_closure *on_resolution =
+      grpc_closure_create(on_resolution_cb, &on_res_arg);
+
+  grpc_resolver_next(&exec_ctx, resolver, &on_res_arg.resolver_result,
+                     on_resolution);
   GRPC_RESOLVER_UNREF(&exec_ctx, resolver, "test_succeeds");
   grpc_uri_destroy(uri);
   grpc_exec_ctx_finish(&exec_ctx);
+  gpr_free(on_res_arg.expected_server_name);
```

Comment:
What guarantee is there that `on_resolution_cb()` will be invoked before this cleanup is done?